### DISCUSSION TITLE
Ignore ResizeObserver errors in the overlay

### DISCUSSION THREAD

**Discussion by just-boris** [vercel/next.js#51551] [Ideas]
### Goals

When a page uses ResizeObserver API, it may occasionally throw "ResizeObserver loop completed with undelivered notifications" (Firefox) or "ResizeObserver loop limit exceeded" (Chrome). The error has weird semantics, it never gets printed to browser console, only gets caught by `window.onerror` listener (used by the error overlay)

This error is a known false alarm: https://github.com/w3c/csswg-drafts/issues/5488

While it may take a long time to change the standards, but existing consumers may be confused by this, because the error may come from a 3rd party library. You can see many examples of affected people in the linked issues here

* https://github.com/w3c/csswg-drafts/issues/5023#ref-issue-848925376
* https://github.com/WICG/resize-observer/issues/38#ref-pullrequest-665074496

A workaround on customer land is possible:

```js
window.addEventListener('error', function(e){
  if (e.message === "ResizeObserver loop completed with undelivered notifications.") {
    console.log(e)
    // prevent React's listener from firing
    e.stopImmediatePropagation();
    // prevent the browser's console error message 
    e.preventDefault();
  }
});
```

but this is suboptimal, because it requires explicit knowledge about this issue and many people keep creating false bug reports.

It would be great if error overlay will skip the known non-error by default

### Non-Goals

n/a

### Background

n/a

### Proposal

It would be great if the error overlay will skip the known non-error by default

**Comment by icyJoseph** [comment]
- Is it possible to trigger this by an actual programmer error? Then I might want to be aware of it
- Is it the same message across browsers?
- Hiding it during dev mode and then seeing at runtime might make some angry, and add to the fire started by hydration errors, which are observed in two different lights in prod and dev modes

**Reply by just-boris** [reply]
> Is it possible to trigger this by an actual programmer error?

The intent of this error is to capture infinite observation loops, but the current logic is too sensitive and it reports valid use-cases as an error too. Here is a trivial repro: https://codesandbox.io/s/resize-observer-error-repro-rmdzyc

All top matches from Google search (at least for me) lead to some sort of way to ignore the error
* https://stackoverflow.com/questions/64238740/how-to-ignore-the-resizeobserver-loop-limit-exceeded-in-testcafe
* https://stackoverflow.com/questions/63020810/what-is-best-way-in-javascript-to-stop-throwing-resizeobserver-loop-limit-excee
* https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded/50387233#50387233
* https://github.com/juggle/resize-observer/issues/103

> Is it the same message across browsers?

Two possible messages

* "ResizeObserver loop completed with undelivered notifications" ([prescribed by the spec](https://www.w3.org/TR/resize-observer/#deliver-resize-error), and followed in all browsers, except Chrome)
* "ResizeObserver loop limit exceeded" (Chrome)

> Hiding it during dev mode and then seeing at runtime might make some angry

The error uses some very peculiar way to dispatch. It is captured by `window.onerror`, but not visible in the browser console. This is why all error-related tools (dev overlays, and telemetry tools like Sentry) display it, but no user in vanilla browser would ever see it.

Because of this semi-visible behavior here I propose to ignore it to make dev experience better.
